### DISCUSSION TITLE
How can we generate a Datamatrix code in base64 (or byte[])

### DISCUSSION THREAD

**Discussion by abennouna12** [micjahn/ZXing.Net#509] [Q&A]
Hello, 

I was using Datamatrix.Net to generate a Datamatrix code. 

Today we have migrated to .NET 7 and this generation doesn't work anymore. 

So I want to use ZXing.Net to generate this Datamatrix but I didn't find documentation about the subject. 

Does anyone have an idea ?

Thanks.

I have started by writing these two lines but I want to convert the result into Base64 strinr or Vyte[] Without using System.Drawing

```
var writer = new DataMatrixWriter();
var image = writer.encode(content, BarcodeFormat.DATA_MATRIX, 300, 300);
```

**Comment by micjahn** [comment]
What should be the content of the base64 string or byte array? Should it be the data of a specific image format like png or jpeg? Or do you need the raw RGB data values?
Perhaps one of the different barcode writer implementations fits your needs, like the ZXing.BarcodeWriterSvg or ZXing.BarcodeWriterPixelData.
